code_climate.rb

```diff
@@ -0,0 +1,161 @@
+      'rating' => snapshot['attributes']['ratings'].first['letter']
+    }
+  end
+
+  def repo_from_github_slug(slug)
+    query = { "github_slug" => slug }
+    path = "/v1/repos?#{query.to_query}"
+    response = CodeClimateRequest.new.request(path)
+    response_to_data(response, path)&.first
+  end
+
+  def repo_from_id(id)
+    path = "/v1/repos/#{id}"
+    response = CodeClimateRequest.new.request(path)
+    response_to_data(response, path)&.first
+  end
+
+  def services(repo_id)
+    path = "/v1/repos/#{repo_id}/services"
+    response = CodeClimateRequest.new.request(path)
+    response_to_data(response, path)
+  end
+
+  def snapshot(repo_id, snapshot_id)
+    path = "/v1/repos/#{repo_id}/snapshots/#{snapshot_id}"
+    response = CodeClimateRequest.new.request(path)
+    response_to_data(response, path)
+  end
+
+  def issues(repo_id, snapshot_id)
+    multi_paged_query("/v1/repos/#{repo_id}/snapshots/#{snapshot_id}/issues")
+  end
+
+  def files(repo_id, snapshot_id)
+    multi_paged_query("/v1/repos/#{repo_id}/snapshots/#{snapshot_id}/files")
+  end
+
+  private
+
+  def multi_paged_query(base_path)
+    result = []
+    page_number = 0
+    query = {
+      "page[size]"   => 100,
+      "page[number]" => page_number
+    }
+    
+    while true
+      page_number += 1
+      query["page[number]"] = page_number
+      path = "#{base_path}?#{query.to_query}"
+      response = CodeClimateRequest.new.request(path)
+      data = response_to_data(response, path)
+      break if data.nil?
+      result += data
+    end
+    
+    result
+  end
+
+  def response_to_data(response, path = nil)
+    data = nil
+
+    if response.code == "200"
+      parsed_response = JSON.parse(response.body)
+      if parsed_response["data"].size > 0
+        data = parsed_response["data"]
+      else
+        if ENV["DEBUG"] == "1"
+          STDERR.puts "Invalid response for path=\"#{path}\": #{parsed_response.inspect}"
+        end
+      end
+    else
+      STDERR.puts "Invalid response code for path=\"#{path}\": #{response.code}"
+    end
+    
+    data
+  end
+end
+
+ALL_REPO_NAMES = [
+  "ManageIQ/manageiq",
+  "ManageIQ/manageiq-schema",
+  "ManageIQ/manageiq-api",
+  "ManageIQ/manageiq-ui-classic",
+  "ManageIQ/manageiq-ui-service",
+  "ManageIQ/manageiq-automation_engine",
+  "ManageIQ/manageiq-content",
+  "ManageIQ/manageiq-providers-amazon",
+  "ManageIQ/manageiq-providers-ansible_tower",
+  "ManageIQ/manageiq-providers-azure",
+  "ManageIQ/manageiq-providers-azure_stack",
+  "ManageIQ/manageiq-providers-foreman",
+  "ManageIQ/manageiq-providers-google",
+  "ManageIQ/manageiq-providers-kubernetes",
+  "ManageIQ/manageiq-providers-lenovo",
+  "ManageIQ/manageiq-providers-nuage",
+  "ManageIQ/manageiq-providers-openshift",
+  "ManageIQ/manageiq-providers-openstack",
+  "ManageIQ/manageiq-providers-ovirt",
+  "ManageIQ/manageiq-providers-redfish",
+  "ManageIQ/manageiq-providers-scvmm",
+  "ManageIQ/manageiq-providers-vmware",
+]
```

Comment:
This bit makes me think this should ultimately live in manageiq-release, but this is good for now :+1: